### ISSUE TITLE
Questions about the hyperparameters and evaluation metric.

### ISSUE THREAD

**Issue by desmond1687** [ucasYW/Contextual-Interaction-for-AQA#2]
Thank you for such excellent work!
Would like to ask you a few questions:
1. Is this set of hyperparameters in `run.sh` optimal in your experiments? Is `run.sh` the `RoBERTa w/ DAGN and CL` in Table 1?
2. I did not find the evaluation metric functions in the code, i.e. pearson and spearman. Do I need to implement them myself?
3. There is a bug in `dagn.py` Line 464. Do I need to change `loss2 = self.get_con_loss(positive_keys2, negative_keys2)` into `loss2 = self.get_con_loss(query2, positive_keys2, negative_keys2)` or `loss2 = self.get_con_lossL(positive_keys2, negative_keys2)`?

Thanks a lot!

**Comment by ucasYW** [comment]
Those hyperparameters  should be optimal, but you can try other parameters to see if you can achieve a better performance, and yes.
For the evaluation, I use corrcoef from numpy and pearson from scipy to calculate them. 
For the 2 different constrastive loss function, use the loss2 = self.get_con_loss(query2, positive_keys2, negative_keys2) will have a better performance, same for the loss3.
Sorry for my poor coding, I planned to make the code more elegant before uploading, but I haven't done it yet

**Comment by desmond1687** [comment]
@ucasYW Thanks for your reply!
However, there seems to be a bug in the `get_con_loss` function:
In `dagn.py`, line 120, `loss =+ nce_fct(query.unsqueeze(0), positive_keys[i].unsqueeze(0), negative_keys)`. Should the "=+" here be changed to "+="? In addition, it is necessary to initialize the loss to 0 before this loop.

**Comment by ucasYW** [comment]
To be honest, that should be a bug but it just works. And the get_con_loss function is the very beginning version of our contrastive loss design, the get_con_lossL() is strictly followed the method we present in the paper. Use the get_con_loss will give you little improvement in correlation metrics, so I still keep it.

**Comment by desmond1687** [comment]
Thanks a lot!
